test/e2e/storage/testsuites/subpath.go

```diff
@@ -587,16 +586,35 @@ func testPodFailSubpathError(f *framework.Framework, pod *v1.Pod, errorMsg strin
 	defer func() {
 		framework.DeletePodWithWait(f, f.ClientSet, pod)
 	}()
+	By("Checking for subpath error in container status")
+	err = waitForPodSubpathError(f, pod)
+	Expect(err).NotTo(HaveOccurred(), "while waiting for subpath failure")
+}
 
-	By("Checking for subpath error event")
-	selector := fields.Set{
-		"involvedObject.kind":      "Pod",
-		"involvedObject.name":      pod.Name,
-		"involvedObject.namespace": f.Namespace.Name,
-		"reason":                   "Failed",
-	}.AsSelector().String()
-	err = framework.WaitTimeoutForPodEvent(f.ClientSet, pod.Name, f.Namespace.Name, selector, errorMsg, framework.PodStartTimeout)
-	Expect(err).NotTo(HaveOccurred(), "while waiting for failed event to occur")
+func waitForPodSubpathError(f *framework.Framework, pod *v1.Pod) error {
+	return wait.PollImmediate(framework.Poll, framework.PodStartTimeout, func() (bool, error) {
+		pod, err := f.ClientSet.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, status := range pod.Status.ContainerStatuses {
+			// 0 is the container that uses subpath
+			if status.Name == pod.Spec.Containers[0].Name {
+				if status.State.Running != nil {
+					return false, fmt.Errorf("subpath container unexpectedly became running")
+				} else if status.State.Terminated != nil {
+					return false, fmt.Errorf("subpath container unexpectedly terminated")
+				} else if status.State.Waiting != nil {
+					if status.State.Waiting.Reason == "CreateContainerConfigError" &&
+						strings.Contains(status.State.Waiting.Message, "subPath") {
+						return true, nil
+					}
+					return false, nil
+				}
+			}
+		}
+		return false, nil
```

Comment:
This case we never found a status corresponding to the container name. Just want to make sure simply retrying is the right behavior here

Reply:
This is similar to the case where container status may not be initially populated